### ISSUE TITLE
Placing (Start Channel) model at end of base model in inherited layout

### ISSUE THREAD

**Issue by thekameleon** [xLightsSequencer/xLights#4159]
I had tried to set the start channel of a model in Xlights at the end of model in my base layout. While it worked, what I couldn't do was move the base model's start channels. I'd like to see the ability to insert the channel assignments between models in a base layout without modifying the base layout. They would entail overriding the "end of model" property of the following base model.

I have just added the channel assignments at the end of the models of the base layout. The challenge is trying to determine which models are in the base layout and which are in the inherited layout. Maybe change the font color for the base layout items?

**Comment by cybercop23** [comment]
Can you please describe a use case? Not sure I get it. Also... any reason why you aren't using the Visualizer to assign channels? There really shouldn't be any manual channel settings nowadays.

**Comment by thekameleon** [comment]
I have 3 models. 2 of these models (model a nd c) are defined in my base layout. 1 is defined in inherited layout (model b). Physically they are laid out from the port as A, B, then C. When I try to add model B between A and C in my layout. It doesn't move C's start channel to the end of B. It remains at the end of A.

I am not sure why you think there is manual channel setting going on. I'm not using the visualizer as it's giving me more headaches than it has done good.

**Comment by cybercop23** [comment]
If you aren't using the visualizer then you are manually assigning them even if you just choose start/end of model.
Try using the viz and see what happens. Also try, just to make sure it works as expects, to add B to the end of C in visualizer.

**Comment by thekameleon** [comment]
I don't want B at the end of C. I want B between A and C.

**Comment by cybercop23** [comment]
"also try".... try to see what happens...

**Comment by thekameleon** [comment]
Well, I am not sure what was happening.  In the Visualizer, It would only allow B to be before A and C or after A and C.  I could not put B in between. What I hope is a bug, even with B before A and C, or after, in the layout's tab, the B model did not move to before or after the A and C model, which is what I would hope would happen.  I am hoping I am doing something wrong here. At least, I hope the visualizer is syncing back to the properties in the layout tab.

To be frank, I am not a fan of the visualizer.  I have a  very mature light show with well over 1,000 models.  and over 1,000,000 channels.  Anytime I moved something in the visualizer, it would take 30 seconds or so for the UI to update on an I9, 64GB, 3080Ti machine.  Visualizer may be good for smaller shows and newer folks to the hobby. For my show, I estimate it could take 3 to 4 weeks at 60 hours a week to completely embrace the visualizer. Assuming it stops doing weird things, like breaking up strings to different ports or trying to figure out how to span two controllers on one model.

**Comment by cybercop23** [comment]
Wow. That is a lot. So most likely you don't have Auto Layout, Auto Size and Full xLights Control on so it will take a while to set up. I have no doubt it will help you and make thinks easier once you make the initial move.. however...

I just tried the same thing. Outline, Door and Windows are in the Base folder. Window Left and Window Right are daizy chained on Port 1 in base folder. Switched to "show" folder where I have the rest of the models "locally" and put a star in the middle of the windows. See below. You can tell by the channel numbers. So what you are trying to do works if using the Visualizer. I'm also including my base & show folders if you want to try yourself.
[Base_ShowFolder_InsertModelInMiddleOfChain.zip](https://github.com/xLightsSequencer/xLights/files/14077733/Base_ShowFolder_InsertModelInMiddleOfChain.zip)

![image](https://github.com/xLightsSequencer/xLights/assets/23619433/3e4378c4-23ef-46e4-ad7c-bdad7aed61b2)

**Comment by thekameleon** [comment]
Even if this works I'm the visualizer. The visualizer and how the models are ordered by channel in the layouts tab should be in sync.

You are right, the first thing I do when I add a new controller is turn off auto layout, auto size ,etc... I tried it a few times and ninja's been disastrous each time I started to try to go down that road. There are a number of reasons. For instance if I drop a model with a number of strings say like a matrix. It tries to force each string onto a different port. Or what do I do with my megatree that requires two controllers. And no shadow models are not the answer, that's a hack.

Visualizer is a great idea in concept but if it is going to support more complex shows. It needs work... possibly a retooling. 

I know it's not what you want to hear. Just my 2 cents.

**Comment by AzGilrock** [comment]
You defined the matrix wrong then cause number of strings = number of ports in xLights.  I doubt any dev will change code for this one.

**Comment by thekameleon** [comment]
That's cool. I just made a suggestion. As far as the Matrix that may be true when I defined it years ago. I kind of wish I could define models exactly how they are physically configured. My Matrix because of its non rectangular shape (hexagon, kinda like a gable roof) required some non-conventional design and configuration on my controller.

**Comment by cybercop23** [comment]
That's totally supported. Custom Matrix, set String, Individial start channels.. set what you need.
![image](https://github.com/xLightsSequencer/xLights/assets/23619433/7108825a-d956-425f-9000-29e333e9cef7)

**Comment by cybercop23** [comment]
Yeah.. tough one.. if it works in Viz and the new way of doing it and you want to use the old way, I'm affraid you're stuck with that. If you try the latest version and the new way of doing it (we're offering zoom help) and doesn't work, that may be a case. Else, IDK what we can do.
I do have the same issue with lots of groups I'd like to suppress, but I use previews, or better yet map at the submodel level.

**Comment by thekameleon** [comment]
It is the time that is really my biggest concern. With these large shoes, xlights is dog slow when ever you add models, change group names, I wish I knew why and I might change. 

With the visualizer, I experienced the same thing. I waited 15 to 20 seconds to model each model from one port to another. It's CPU cycles, but not sure what's going on.

Another thing, I just thought of and I'm curious what you will suggest. What if i have a port that goes bad, will the expectation that I have to open open xlights chnage the configuration in xlights, then upload to the controller?

I go back to an earlier request where I wish this visualizer was more suggestive / optional and I could ask it to ignore specific ports on a controller. I would make it easier to deal with random customizations and allow me to move over to it on my time.
